pkg/chunk/cache/redis_client.go

```diff
@@ -61,14 +71,49 @@ func NewRedisClient(cfg *RedisConfig) *RedisClient {
 		IdleTimeout: cfg.IdleTimeout,
 		MaxConnAge:  cfg.MaxConnAge,
 	}
+
+	var addresses []string
+	if len(cfg.Addresses) > 0 {
+		fmt.Println("here!")
+		util_log.WarnExperimentalUse("DNS-based redis service discovery")
+
+		ctx, cancel := context.WithTimeout(context.Background(), 10*time.Second)
+		defer cancel()
+
+		resolver := dns.NewResolver(dns.GolangResolverType.ToResolver(logger), logger)
+		for _, addr := range strings.Split(cfg.Addresses, ",") {
+			qtype, name := dns.GetQTypeName(addr)
+			if qtype == "" {
+				addresses = append(addresses, name)
+				continue
+			}
+			resolved, err := resolver.Resolve(ctx, name, dns.QType(qtype))
+			if err != nil {
+				level.Error(logger).Log("msg", "failed to resolve redis address", "addr", addr, "err", err)
```

Comment:
Stylistically, is it better to log here rather than returning the error?
What happens if all the addresses result in error?

Reply:
Good point, is it better to return an error on first address resolve failure or return only if all addresses fail to resolve?

Reply:
I went with the second option to return an error if no addresses are resolved.